model/host/db.go

```diff
@@ -1376,6 +1375,25 @@ func (h *Host) IncrementNumAgentCleanupFailures(ctx context.Context) error {
 				NumAgentCleanupFailuresKey: 1,
 			},
 		},
+	); err != nil {
+		return errors.Wrapf(err, "incrementing number of agent cleanup failures for host '%s'", h.Id)
+	}
+	h.NumAgentCleanupFailures++
+	return nil
+}
+
+// UnsetNumAgentCleanupFailures unsets the NumAgentCleanupFailures field.
+func (h *Host) UnsetNumAgentCleanupFailures(ctx context.Context) error {
+	err := UpdateOne(
+		ctx,
+		bson.M{
+			IdKey: h.Id,
+		},
+		bson.M{
+			"$unset": bson.M{
```

Comment:
Might be slightly better to `$set` it explicitly to 0 rather than delete the field entirely since the data model doesn't have `omitempty` for the BSON struct tags.